src/test/java/com/puppycrawl/tools/checkstyle/JavaAstVisitorTest.java

```diff
@@ -164,6 +165,42 @@ public void testOrderOfVisitMethodsAndProductionRules() throws Exception {
                 .inOrder();
     }
 
+    /**
+     * The reason we have this test is that we forgot to add the imaginary 'EXPR' node
+     * to a production rule in the parser grammar (we should have used 'expression'
+     * instead of 'expr'). This test is a reminder to question the usage of the 'expr' parser
+     * rule in the parser grammar when we update the count to make sure we are not missing
+     * an imaginary 'EXPR' node in the AST.
+     *
+     * @throws IOException if file does not exist
+     */
+    @Test
+    public void countExprUsagesInParserGrammar() throws IOException {
```

Comment:
Hopefully we can prevent this in the future. I had considered using a custom check for this, and a few other things that included individual suppressions, but really we should not need to get this crazy. As long as we can explain why we increase the number of expected usages of the `expr` rule, we should be fine. I placed a comment directly above this value so that we will notice updates to this number and catch it in review.